indico/modules/rb/models/reservations.py

```diff
@@ -417,8 +417,7 @@ def accept(self, user, reason=None):
         self.add_edit_log(ReservationEditLog(user_name=user.full_name, info=[log_msg]))
         notify_confirmation(self, reason)
         signals.rb.booking_state_changed.send(self)
-        valid_occurrences = self.occurrences.filter(ReservationOccurrence.is_valid).all()
-        pre_occurrences = ReservationOccurrence.find_overlapping_with(self.room, valid_occurrences, self.id).all()
+        pre_occurrences = get_prebooking_collisions(self)
         for occurrence in pre_occurrences:
             if not occurrence.is_valid:
```

Comment:
This check should probably be inside `get_prebooking_collisions` - otherwise an already-cancelled prebooking occurrence would also show up as a collision when you call the util from the RH.